integration/monitoring_federation_test.go

```diff
@@ -0,0 +1,252 @@
+package integration_test
+
+import (
+	"context"
+	"fmt"
+	"testing"
+	"time"
+
+	monitoringv1 "github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	corev1 "k8s.io/api/core/v1"
+	k8sApiErrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	addonsv1alpha1 "github.com/openshift/addon-operator/apis/addons/v1alpha1"
+	"github.com/openshift/addon-operator/integration"
+	"github.com/openshift/addon-operator/internal/controllers"
+)
+
+func (s *integrationTestSuite) TestMonitoringFederation() {
+	s.Run("MonitoringInPlaceAtCreationRemovedAfterwards", func() {
```

Comment:
As `MonitoringInPlaceAtCreationRemovedAfterwards` and `MonitoringNotInPlaceAtCreationAddedAfterwards` don't share anything, what about pulling them top level into `func (s *integrationTestSuite) TestMonitoringFederation_MonitoringInPlaceAtCreationRemovedAfterwards()` and `func (s *integrationTestSuite) TestMonitoringFederation_MonitoringNotInPlaceAtCreationAddedAfterwards()`?